drivers/ethernet/eth_gecko.c

```diff
@@ -60,6 +60,29 @@ static void link_configure(ETH_TypeDef *eth, u32_t flags)
 	eth->NETWORKCTRL |= (ETH_NETWORKCTRL_ENBTX | ETH_NETWORKCTRL_ENBRX);
 }
 
+static void eth_gecko_setup_mac(struct device *dev)
+{
+	const struct eth_gecko_dev_cfg *const cfg = DEV_CFG(dev);
+	ETH_TypeDef *eth = cfg->regs;
+	u32_t link_status;
+	int result;
+
+	/* PHY auto-negotiate link parameters */
+	result = phy_gecko_auto_negotiate(&cfg->phy, &link_status);
+	if (result < 0) {
+		LOG_ERR("ETH PHY auto-negotiate sequence failed");
+		return;
+	}
+
+	LOG_INF("Speed %s Mb",
+		link_status & ETH_NETWORKCFG_SPEED ? "100" : "10");
+	LOG_INF("%s duplex",
+		link_status & ETH_NETWORKCFG_FULLDUPLEX ? "Full" : "Half");
+
```

Comment:
It would make sense to add `ETHERNET_AUTO_NEGOTIATION_SET | ETHERNET_DUPLEX_SET` capabilities to `eth_gecko_get_capabilities()` if we support changing these. That can be sent as a separate PR.

Reply:
Agreed. Because this PR changes this capability I have included it in this PR.